find_flaky_tests.py

```diff
@@ -120,53 +121,66 @@ def render_msg_header(state: AppState) -> str:
 
 
 def print_for_slack(occurrences: List[Occurrence], state: AppState):
-    json = """
-    {
-        "channel": "@CHANNEL@",
+    occurrences_by_ann_path: Dict[str, List[Occurrence]] = {}
+    for o in occurrences:
+        occurrences_by_ann_path.setdefault(o.annotation_path, []).append(o)
+    items = [i for i in occurrences_by_ann_path.items()]
+    items.sort(key=lambda i: len(i[1]), reverse=True)  # sort by number of occurrences, highest first
+
+    limit = 12
+    items = items[:limit]
+
+    # Chunk results so that individual message blocks do not exceed the 3000 character limit from Slack API
+    chunk_size=5
+    chunks = [items[i:i+chunk_size] for i in range(0, len(items), chunk_size)]
+
+    content_blocks = []
+
+    for chunk in chunks:
+        content = ""
+        for ann_path, occrs in chunk:
+            nice_path = truncate_left(ann_path, MAX_FILENAME_LENGTH)
+            count = len(occrs)
+            content += f"{count}x `{nice_path}` "
+            occrs.sort(key=lambda o: o.commit.timestamp, reverse=True)
+            recent_occrs = occrs[:5]
+            content += " ".join([f"<{occr.check_url}|{i+1}> " for i, occr in enumerate(recent_occrs)])
+            content += '\n'
+
+        content_blocks += [
+            {
+                "type": "section",
+                "text": {
+                    "type": "mrkdwn",
+                    "text": content,
+                }
+            }
+        ]
+
+
+    data = {
+        "channel": state.slack_channel,
```

Comment:
Instead of working with a static string and doing replacements there, we now have a data structure that gets serialized to json later